app/vmstorage/servers/vmselect.go

```diff
@@ -249,10 +250,23 @@ func getMaxMetrics(sq *storage.SearchQuery) int {
 	maxMetrics := sq.MaxMetrics
 	maxMetricsLimit := *maxUniqueTimeseries
 	if maxMetricsLimit <= 0 {
-		maxMetricsLimit = 2e9
+		maxMetricsLimit = calculateMaxMetricsLimitByResource(*maxConcurrentRequests, memory.Remaining())
```

Comment:
I think this should be logged on process startup, similarly to how we do with mem:
```
limiting caches to 20615843020 bytes, leaving 13743895348 bytes to the OS according to -memory.allowedPercent=60

```

Reply:
Would be also great if we can expose this in a form of a metric as well.